processing/src/main/java/io/druid/segment/incremental/OakIncrementalIndex.java

```diff
@@ -0,0 +1,569 @@
+      public Object apply(Map.Entry<ByteBuffer, ByteBuffer> entry)
+      {
+        ByteBuffer serializedValue = entry.getValue();
+        BufferAggregator agg = getAggs()[aggIndex];
+        return agg.get(serializedValue, serializedValue.position() + aggsManager.aggOffsetInBuffer[aggIndex]);
+      }
+    };
+
+    OakTransformView<IncrementalIndexRow, Object> transformView = (OakTransformView<IncrementalIndexRow, Object>) oak.createTransformView(transformer);
+    return transformView.get(incrementalIndexRow);
+  }
+
+  @Override
+  protected float getMetricFloatValue(IncrementalIndexRow incrementalIndexRow, int aggIndex)
+  {
+    Function<Map.Entry<ByteBuffer, ByteBuffer>, Float> transformer = new Function<Map.Entry<ByteBuffer, ByteBuffer>, Float>() {
+      @Override
+      public Float apply(Map.Entry<ByteBuffer, ByteBuffer> entry)
+      {
+        ByteBuffer serializedValue = entry.getValue();
+        BufferAggregator agg = getAggs()[aggIndex];
+        return agg.getFloat(serializedValue, serializedValue.position() + aggsManager.aggOffsetInBuffer[aggIndex]);
+      }
+    };
+
+    OakTransformView<IncrementalIndexRow, Float> transformView = (OakTransformView<IncrementalIndexRow, Float>) oak.createTransformView(transformer);
+    return transformView.get(incrementalIndexRow);
+  }
+
+  @Override
+  protected long getMetricLongValue(IncrementalIndexRow incrementalIndexRow, int aggIndex)
+  {
+    Function<Map.Entry<ByteBuffer, ByteBuffer>, Long> transformer = new Function<Map.Entry<ByteBuffer, ByteBuffer>, Long>() {
+      @Override
+      public Long apply(Map.Entry<ByteBuffer, ByteBuffer> entry)
+      {
+        ByteBuffer serializedValue = entry.getValue();
+        BufferAggregator agg = getAggs()[aggIndex];
+        return agg.getLong(serializedValue, serializedValue.position() + aggsManager.aggOffsetInBuffer[aggIndex]);
+      }
+    };
+
+    OakTransformView<IncrementalIndexRow, Long> transformView = (OakTransformView<IncrementalIndexRow, Long>) oak.createTransformView(transformer);
+    return transformView.get(incrementalIndexRow);
+  }
+
+  @Override
+  protected Object getMetricObjectValue(IncrementalIndexRow incrementalIndexRow, int aggIndex)
+  {
+    Function<Map.Entry<ByteBuffer, ByteBuffer>, Object> transformer = new Function<Map.Entry<ByteBuffer, ByteBuffer>, Object>() {
+      @Override
+      public Object apply(Map.Entry<ByteBuffer, ByteBuffer> entry)
+      {
+        ByteBuffer serializedValue = entry.getValue();
+        BufferAggregator agg = getAggs()[aggIndex];
+        return agg.get(serializedValue, serializedValue.position() + aggsManager.aggOffsetInBuffer[aggIndex]);
+      }
+    };
+
+    OakTransformView<IncrementalIndexRow, Object> transformView = (OakTransformView<IncrementalIndexRow, Object>) oak.createTransformView(transformer);
+    return transformView.get(incrementalIndexRow);
+  }
+
+  @Override
+  protected double getMetricDoubleValue(IncrementalIndexRow incrementalIndexRow, int aggIndex)
```

Comment:
Very similar code everywhere, but can't see how to eliminate the duplication, the getTTT method is different everywhere.